src/core/p5.Renderer.js

```diff
@@ -120,26 +120,28 @@ class Renderer extends p5.Element {
   /**
  * Resize our canvas element.
  */
-  resize (w, h) {
-    this.width = w;
-    this.height = h;
-    this.elt.width = w * this._pInst._pixelDensity;
-    this.elt.height = h * this._pInst._pixelDensity;
-    this.elt.style.width = `${w}px`;
-    this.elt.style.height = `${h}px`;
-    if (this._isMainCanvas) {
-      this._pInst._setProperty('width', this.width);
-      this._pInst._setProperty('height', this.height);
+  resize(w, h) {
+    if (this._pInst && this._pInst._pixelDensity) {
```

Comment:
Was there a case where these conditions weren't met?

Reply:
Yes, during testing, there seems to be an issue where, after a resize operation, logging `console.log(this._pInst)` sometimes results in the value `undefined`. This occurrence leads to errors of the form "cannot read properties of undefined (reading `_pixelDensity`)". To address this, a check has been added to verify whether `this._pInst` is defined before attempting to access its properties. This check helps prevent the errors encountered during the testing process. It was actually failing the test given below. Am I doing the correct way....Or this still have limitations?
```js
 test('secondary graphics layer matches main canvas size', function() {
      let g1 = myp5.createCanvas(5, 5, myp5.WEBGL);
      let s = myp5.createShader(vert, frag);
      myp5.filter(s);
      let g2 = g1.filterGraphicsLayer;
      assert.deepEqual([g1.width, g1.height], [g2.width, g2.height]);
      myp5.resizeCanvas(4, 4);
      assert.deepEqual([g1.width, g1.height], [g2.width, g2.height]);
    });
```
![checkpass](https://github.com/processing/p5.js/assets/127239756/df1026d8-c2da-4939-9a97-0aec445b754e)

Reply:
thanks for pointing out the test case! I was asking because it seems like it might be indicative of a bigger problem if something is getting resized without having a `_pInst`, and maybe we want to see how it got into that state and maybe fix it at the source instead of here. I'll take a look at that test and get back to you soon!

Reply:
Sure

> thanks for pointing out the test case! I was asking because it seems like it might be indicative of a bigger problem if something is getting resized without having a `_pInst`, and maybe we want to see how it got into that state and maybe fix it at the source instead of here. I'll take a look at that test and get back to you soon!

Sure

Reply:
I think it's coming from these lines: https://github.com/processing/p5.js/blob/28740f9cda2e6983bd27c0564cd029fa9241d733/src/webgl/p5.RendererGL.js#L1432-L1435

These were added when `filterGraphicsLayer` was a graphic, but now it's a framebuffer. I think we don't need those lines at all any more since we handle the resizing when we call `filter`, so maybe we can take it out, and check instead that the size changes after calling `filter`:

```diff
    test('secondary graphics layer matches main canvas size', function() {
      let g1 = myp5.createCanvas(5, 5, myp5.WEBGL);
      let s = myp5.createShader(vert, frag);
      myp5.filter(s);
      let g2 = g1.filterGraphicsLayer;
      assert.deepEqual([g1.width, g1.height], [g2.width, g2.height]);
      myp5.resizeCanvas(4, 4);
+     myp5.filter(s);
      assert.deepEqual([g1.width, g1.height], [g2.width, g2.height]);
    });
```

Also as a side note, we should still rename `filterGraphicsLayer` and `filterGraphicsLayerTemp` in `RendererGL` to remove the word `graphic` (just `filterLayer` and `filterLayerTemp` probably)

Reply:
So, i have done a little testing myself, I think it's working now. I have made the required changes.